### PR TITLE
Upgrade to MongoDb 7

### LINKED ISSUE #200: Upgrade to MongoDb 7
The CI as well as the local development and production `docker-compose.yml` files should be updated to use MongoDB 7.

### PR DESCRIPTION
**Pull request by VKTB**
## Description
Updates the e2e CI job and `docker-compose.yml` file to use MongoDB 7.

## Testing instructions
- [x] Review code
- [x] Check Actions build
- [x] Review changes to test coverage
- [x] Check app runs fine
- [x] Check that importing the units works

## Agile board tracking
closes #200

### CODE REVIEW

File: .github/workflows/.ci.yml [hunk 1]
```diff
@@ -56,7 +56,7 @@ jobs:
         run: pytest -c test/pytest.ini test/unit/ --cov
 
   e2e-tests:
-    needs: [unit-tests]
+    needs: [ unit-tests ]
```
Comment (joelvdavies) [inline .github/workflows/.ci.yml:59]: Have you got a different formatter to me? Mine just removes these spaces again. Its only this line and 94 that conflict with mine. I personally prefer not having spaces like this.

**Review by joelvdavies — COMMENTED** [review]
A comment on formatting. Also my tests cant discover because of a permission error on `/home/qha22535/inventory-management-system-api/mongodb/data/diagnostic.data/__init__.py` when I am using
```
"python.testing.pytestArgs": [
  "-c",
  "test/pytest.ini",
  "test"
]
```
in my vscode config. I tried deleting the whole directory after testing it worked with my existing data but it didn't fix it.

File: .github/workflows/.ci.yml (hunk 1 above)
Comment (VKTB) [reply .github/workflows/.ci.yml:59]: PyCharm's formatter, and I think you use VS Code?

Comment (VKTB) [inline .github/workflows/.ci.yml:59]:
```suggestion
    needs: [unit-tests]
```

File: .github/workflows/.ci.yml [hunk 2]
```diff
@@ -91,31 +91,31 @@ jobs:
   docker:
     # This job triggers only if all the other jobs succeed. It builds the Docker image and if successful,
     # it pushes it to Harbor.
-    needs: [linting, unit-tests, e2e-tests]
+    needs: [ linting, unit-tests, e2e-tests ]
```
Comment (VKTB) [inline .github/workflows/.ci.yml:94]:
```suggestion
    needs: [linting, unit-tests, e2e-tests]
```

**Review by joelvdavies — APPROVED** [review]

**Pull request merged**